<launch>
    <!-- STEREO SPINNAKER NODE PARAMETERS -->
    <arg name="camera_name"               default="stereo" />
    <arg name="run_stereo_image_proc"     default="false" />

    <!-- DJI SDK NODE -->
    <node pkg="dji_osdk_ros" type="dji_vehicle_node" name="vehicle_node" output="screen">
        <param name="acm_name" type="string" value="/dev/ttyACM0"/>
        <param name="serial_name" type="string" value="/dev/ttyTHS1"/>
        <param name="baud_rate" type="int" value="1000000"/>
        <param name="app_id" type="int" value="1076017"/>
        <param name="app_version" type="int" value="1"/>
        <param name="align_time" type="bool" value="false"/>
        <param name="enc_key" type="string" value="6bd1d26f8dd897e4b6864aebd9fce5fa41070d27d2e63867f2b18c598e6c2fc3"/>
        <param name="use_broadcast" type="bool" value="false"/>
    </node>

    <!-- SPINNAKER STEREO NODE -->
  <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" />
    <group ns="$(arg camera_name)" >
        <group ns="left" >
            <!-- Camera nodelet -->
            <node pkg="nodelet" type="nodelet" name="camera_nodelet"
                    args="load spinnaker_camera_driver/SpinnakerCameraNodelet /camera_nodelet_manager" >
                <param name="frame_id"                        value="camera_left" />
                <param name="serial"                          value="19290922" />

                <param name="acquisition_frame_rate_enable"   value="true" />
                <param name="acquisition_frame_rate"          value="15" />
                <param name="image_format_color_coding"       value="BayerRG8"/>
                <param name="auto_white_balance"              value="Continuous"/>
                <param name="exposure_auto"                   value="Once"/>
                <param name="exposure_time"                   value="100"/>
                <param name="auto_exposure_time_upper_limit"  value="5000"/>
                <param name="white_balance_blue_ratio"        value="800"/>
                <param name="white_balance_red_ratio"         value="550"/>
            </node>
            <!-- Debayering nodelet -->
            <node pkg="nodelet" type="nodelet" name="image_proc_debayer"
                args="load image_proc/debayer /camera_nodelet_manager">
            </node>
        </group>
        <group ns="right" >
            <!-- Camera nodelet -->
            <node pkg="nodelet" type="nodelet" name="camera_nodelet"
                    args="load spinnaker_camera_driver/SpinnakerCameraNodelet /camera_nodelet_manager" >
                <param name="frame_id"                        value="camera_right" />
                <param name="serial"                          value="19355648" />

                <param name="acquisition_frame_rate_enable"   value="true" />
                <param name="acquisition_frame_rate"          value="15" />
                <param name="image_format_color_coding"       value="BayerRG8"/>
                <param name="auto_white_balance"              value="Continuous"/>
                <param name="exposure_auto"                   value="Once"/>
                <param name="exposure_time"                   value="100"/>
                <param name="auto_exposure_time_upper_limit"  value="5000"/>
                <param name="white_balance_blue_ratio"        value="800"/>
                <param name="white_balance_red_ratio"         value="550"/>
            </node>
            <!-- Debayering nodelet -->
            <node pkg="nodelet" type="nodelet" name="image_proc_debayer"
                args="load image_proc/debayer /camera_nodelet_manager">
            </node>
        </group>
        <!-- Stereo image processing nodelet -->
        <group if="$(arg run_stereo_image_proc)" >
            <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc">
                <param name="approximate_sync" value="true"/>
            </node>
        </group>
    </group>
    <!-- V3D STEREO NODE -->
    <node pkg="stereo_vant" type="stereo_vant3d_node" name="stereo_vant3d" output="screen">
        <param name="gps_topic" type="string" value="/dji_osdk_ros/gps_position" />
        <param name="rtk_topic" type="string" value="/dji_osdk_ros/gps_position" />
        <param name="imu_topic" type="string" value="/dji_osdk_ros/imu" />
        <param name="camera_left_topic" type="string" value="/stereo/left/image_raw" />
        <param name="camera_right_topic" type="string" value="/stereo/right/image_raw" />
    </node>
</launch>